addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/internal/state/MonitorThingState.java

```diff
@@ -0,0 +1,537 @@
+            return ZoneMinderMonitorStatusEnum.UNKNOWN;
+        }
+
+        try {
+            result = ZoneMinderMonitorStatusEnum.getEnumFromName(stateStatus.toString());
+        } catch (Exception ex) {
+            logger.error(
+                    "{}: context='getMonitorDetailedStatus' Exception occurred when calling getMonitorDetailedStatus",
+                    logIdentifier, ex);
+
+        }
+        return result;
+    }
+
+    public void setMonitorTriggerEvent(ZoneMinderTriggerEvent event) {
+        isDirty = true;
+        curTriggerEvent = event;
+    }
+
+    public void setMonitorFunction(ZoneMinderMonitorFunctionEnum monitorFunction) {
+        GenericChannelState gcs = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_FUNCTION);
+        if (gcs != null) {
+            try {
+                gcs.setState(monitorFunction.toString());
+            } catch (UnsupportedDataTypeException e) {
+                logger.error("{}: context='setMonitorFunction' Exception occurred when updating capture-daemon channel",
+                        logIdentifier, e);
+            }
+        }
+    }
+
+    public void setMonitorEventCause(State state) {
+        GenericChannelState gcs = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_EVENT_CAUSE);
+        if (gcs != null) {
+            try {
+                gcs.setState(state);
+            } catch (UnsupportedDataTypeException e) {
+                logger.error(
+                        "{}: context='setMonitorEventCause' Exception occurred when updating capture-daemon channel",
+                        logIdentifier, e);
+
+            }
+        }
+
+    }
+
+    public void setMonitorEventMotion(State state) {
+        GenericChannelState gcs = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_MOTION_EVENT);
+
+        if (gcs != null) {
+            try {
+                gcs.setState(state);
+            } catch (UnsupportedDataTypeException e) {
+                logger.error("{}: context='setMonitorEnabled' Exception occurred when updating capture-daemon channel",
+                        logIdentifier, e);
+            }
+        }
+
+    }
+
+    public void setMonitorGeneralData(IMonitorDataGeneral monitorData) {
+        if (monitorData != null) {
+            setMonitorFunction(monitorData.getFunction());
+            setMonitorEnabled(monitorData.getEnabled());
+        }
+    }
+
+    public void setMonitorEnabled(boolean state) {
+        GenericChannelState gcs = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_ENABLED);
+
+        try {
+            if (gcs != null) {
+                gcs.setState(state);
+            }
+        } catch (UnsupportedDataTypeException e) {
+            logger.error("{}: context='setMonitorEnabled' Exception occurred when calling setMonitorEnabled",
+                    logIdentifier, e);
+
+        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.